sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -70,24 +70,46 @@ export class BatchingReceiver extends MessageReceiver {
   receive(maxMessageCount: number, idleTimeoutInSeconds?: number): Promise<ServiceBusMessage[]> {
     throwErrorIfConnectionClosed(this._context.namespace);
 
-    if (idleTimeoutInSeconds == undefined) {
+    if (idleTimeoutInSeconds == null) {
       idleTimeoutInSeconds = Constants.defaultOperationTimeoutInSeconds;
     }
 
     const brokeredMessages: ServiceBusMessage[] = [];
 
     this.isReceivingMessages = true;
     return new Promise<ServiceBusMessage[]>((resolve, reject) => {
-      let onReceiveMessage: OnAmqpEventAsPromise;
-      let onSessionClose: OnAmqpEventAsPromise;
-      let onReceiveClose: OnAmqpEventAsPromise;
-      let onReceiveDrain: OnAmqpEvent;
-      let onReceiveError: OnAmqpEvent;
-      let onSessionError: OnAmqpEvent;
       let firstMessageWaitTimer: NodeJS.Timer | undefined;
 
+      const onSessionError: OnAmqpEvent = (context: EventContext) => {
+        this.isReceivingMessages = false;
+        const receiver = this._receiver || context.receiver!;
+        receiver.removeListener(ReceiverEvents.receiverError, onReceiveError);
+        receiver.removeListener(ReceiverEvents.message, onReceiveMessage);
+        receiver.removeListener(ReceiverEvents.receiverDrained, onReceiveDrain);
+        receiver.session.removeListener(SessionEvents.sessionError, onSessionError);
+
+        const sessionError = context.session && context.session.error;
+        let error = new MessagingError("An error occuured while receiving messages.");
+        if (sessionError) {
+          error = translate(sessionError);
+          log.error(
+            "[%s] 'session_close' event occurred for Receiver '%s' received an error:\n%O",
+            this._context.namespace.connectionId,
+            this.name,
+            error
+          );
+        }
+        if (firstMessageWaitTimer) {
+          clearTimeout(firstMessageWaitTimer);
+        }
+        if (this._newMessageReceivedTimer) {
+          clearTimeout(this._newMessageReceivedTimer);
+        }
+        reject(error);
+      };
+
       // Final action to be performed after maxMessageCount is reached or the maxWaitTime is over.
-      const finalAction = () => {
+      const finalAction = (): void => {
```

Comment:
@ramya-rao-a  what will be type annotation for finalAction ?

Reply:
It should return void